genesis/utils/array_class.py

```diff
@@ -20,6 +20,10 @@
 DATA_ORIENTED = partial(dataclasses.dataclass, frozen=True) if gs.use_ndarray else ti.data_oriented
 
 
+def _shape(shape, is_on):
```

Comment:
I would be better to have a more explicit name if you want to keep this pattern. Like `optional_shape` or `cond_shape` or `maybe_shape`.